rvm/src/org/jikesrvm/scheduler/RVMThread.java

```diff
@@ -4023,6 +4023,10 @@ public static void unblockAllMutatorsForGC() {
       RVMThread.handshakeThreads[i] = null; // Help GC
     }
     RVMThread.handshakeLock.unlock();
+
+    if (VM.BuildWithRustMMTk) {
```

Comment:
What about we have an upcall into the VM from Rust when the `FinalizableProcessor` finishes, it schedules the `FinalizerThread` there. That way it's closer to what's done in Java MMTk.

Although I don't have a strong opinion about this, it seems very convenient to schedule the FinalizerThread when unblocking mutators, unless there might be a problem with concurrent GC?

Reply:
Yeah. I will add a method `schedule_finalization()` to one of the VM traits.